src/redturtle/gallery/browser/static/js/integration.js

```diff
@@ -43,25 +45,115 @@ require(['jquery'], function($) {
       }
     });
 
+    $(document).on('keydown', function(e) {
+      if (e.which === 27) {
+        $('.gallery-modal').remove();
+      } else if (e.which === 37) {
+        $('.slick-prev').click();
+      } else if (e.which === 39) {
+        $('.slick-next').click();
+      }
+    });
+
     setTimeout(function() {
       $('.gallery-slider .gallery-item.slick-active').focus();
-    }, 500);
+    }, 0);
   }
 
-  function checkSliderLoaded() {
-    if (
-      $('.gallery-slider').length &&
-      $('.gallery-slider').hasClass('slick-initialized')
-    ) {
-      init();
+  function checkSliderLoaded(slider, callback) {
+    if ($(slider).length && $(slider).hasClass('slick-initialized')) {
+      callback();
     } else {
       setTimeout(checkSliderLoaded, 200);
     }
   }
 
-  $(document).on('ready', function() {
-    $('body').on('init', '.gallery-slider', function() {
-      checkSliderLoaded();
+  function getTemplate(selector) {
+    return $('#' + selector)
+      .html()
+      .replace(/&lt;/g, '<')
+      .replace(/&gt;/g, '>')
+      .replace(/&amp;/g, '&');
+  }
+
+  function createModal(clickedEl, callback) {
+    var elements = $('.photo-gallery .photo-gallery-item');
+
+    var modal = $('<div/>');
+    modal.addClass('gallery-modal');
+    modal.attr('role', 'dialog');
+
+    var contentStructure = getTemplate('photo-gallery-template');
+    modal.html(contentStructure);
+
+    var imgTemplate = getTemplate('photo-gallery-item-template');
+
+    var title = $('.photo-gallery').attr('data-gallery-title');
+
+    modal.find('.gallery-modal-title span').text(title);
+
+    var elementsSrc = [];
+
+    for (var i = 0; i < elements.length; i++) {
+      var img = elements.find('img')[i];
+      elementsSrc.push(elements.find('img')[i].src);
+
+      var el = $('<div/>');
+      el.addClass('gallery-item');
+      el.html(imgTemplate);
+
+      var elImg = el.find('img')[0];
+      elImg.src = img.src;
+      elImg.alt = img.alt;
```

Comment:
Didn't even know that accessing attributes like this would work. For the sake of jqueryness, what do you think of something like:
```javascript
var $elImg = el.find('img');
$elImg.attr({
  src: $img.attr('src'), // $img declared in my comment above
  alt: $img.attr('alt'),
});